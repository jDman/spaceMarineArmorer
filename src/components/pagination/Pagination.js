import React from 'react';

import Button from '../button/Button';

import classes from './Pagination.module.css';

const Pagination = (props) => {
  return (
    <div className={classes.Pagination}>
      {props.hasPrevPage ? (
        <Button
          buttonText="previous"
          classes={classes.PaginationButton}
          clickHandler={() => props.prevClickHandler(props.currentPage)}
        />
      ) : null}
      <p>Page {props.currentPage}</p>
      {props.hasNextPage ? (
        <Button
          buttonText="next"
          classes={classes.PaginationButton}
          clickHandler={() => props.nextClickHandler(props.currentPage)}
        />
      ) : null}
    </div>
  );
};

export default Pagination;
