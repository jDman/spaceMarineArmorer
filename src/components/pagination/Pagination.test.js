import React from 'react';

import renderer from 'react-test-renderer';

import Pagination from './Pagination';

describe('<Pagination />', () => {
  it('should render with both hasPrevPage and hasNextPage', () => {
    const component = renderer.create(
      <Pagination
        hasPrevPage={true}
        hasNextPage={true}
        currentPage={1}
        nextClickHandler={() => {}}
        prevClickHandler={() => {}}
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render with just hasPrevPage', () => {
    const component = renderer.create(
      <Pagination
        hasPrevPage={true}
        hasNextPage={false}
        currentPage={1}
        nextClickHandler={() => {}}
        prevClickHandler={() => {}}
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render with just hasNextPage', () => {
    const component = renderer.create(
      <Pagination
        hasPrevPage={false}
        hasNextPage={true}
        currentPage={1}
        nextClickHandler={() => {}}
        prevClickHandler={() => {}}
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
