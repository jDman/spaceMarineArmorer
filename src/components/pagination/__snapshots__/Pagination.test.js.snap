// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> should render with both hasPrevPage and hasNextPage 1`] = `
<div
  className="Pagination"
>
  <button
    className="PaginationButton"
    onClick={[Function]}
    type="button"
  >
    previous
  </button>
  <p>
    Page 
    1
  </p>
  <button
    className="PaginationButton"
    onClick={[Function]}
    type="button"
  >
    next
  </button>
</div>
`;

exports[`<Pagination /> should render with just hasNextPage 1`] = `
<div
  className="Pagination"
>
  <p>
    Page 
    1
  </p>
  <button
    className="PaginationButton"
    onClick={[Function]}
    type="button"
  >
    next
  </button>
</div>
`;

exports[`<Pagination /> should render with just hasPrevPage 1`] = `
<div
  className="Pagination"
>
  <button
    className="PaginationButton"
    onClick={[Function]}
    type="button"
  >
    previous
  </button>
  <p>
    Page 
    1
  </p>
</div>
`;
